module/applications/advancement/subclass-flow.mjs

```diff
@@ -1,13 +1,133 @@
+    });
+  }
+
+  /* -------------------------------------------- */
+
   /** @inheritDoc */
+  async retainData(data) {
+    await super.retainData(data);
+    const uuid = foundry.utils.getProperty(data, "flags.dnd5e.sourceId");
+    if ( uuid ) this.subclass = await fromUuid(uuid);
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
   async getData() {
-    return foundry.utils.mergeObject(super.getData(), {
-      hint: `${this.advancement.hint ?? ""} ${game.i18n.localize("DND5E.ADVANCEMENT.Subclass.FlowHint")}`
-    });
+    const context = await super.getData();
+    context.subclass = this.subclass;
+    return context;
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  activateListeners(jQuery) {
+    super.activateListeners(jQuery);
+    const [html] = jQuery;
+    html.querySelector("a[data-uuid]")?.addEventListener("click", this._onClickFeature.bind(this));
+    html.querySelector('[data-action="browse"]')?.addEventListener("click", this._onBrowseCompendium.bind(this));
+    html.querySelector('[data-action="deleteItem"]')?.addEventListener("click", this._onItemDelete.bind(this));
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Handle opening the compendium browser and displaying the result.
+   * @param {Event} event  The originating click event.
+   * @protected
+   */
+  async _onBrowseCompendium(event) {
+    event.preventDefault();
+    const filters = {
+      locked: {
+        additional: { class: { [this.item.identifier]: 1 } },
+        types: new Set(["subclass"])
+      }
+    };
+    const result = await CompendiumBrowser.selectOne({ filters });
+    if ( result ) this.subclass = await fromUuid(result);
+    this.render();
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Handle clicking on a feature during item grant to preview the feature.
+   * @param {MouseEvent} event  The triggering event.
+   * @protected
+   */
+  async _onClickFeature(event) {
+    event.preventDefault();
+    const uuid = event.currentTarget.dataset.uuid;
+    const item = await fromUuid(uuid);
+    item?.sheet.render(true);
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Handle deleting a dropped item.
+   * @param {Event} event  The originating click event.
+   * @protected
+   */
+  async _onItemDelete(event) {
+    event.preventDefault();
+    this.subclass = false;
+    this.render();
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritdoc */
+  async _onDrop(event) {
+    // Try to extract the data
+    let data;
+    try {
+      data = JSON.parse(event.dataTransfer.getData("text/plain"));
+    } catch(err) {
+      return false;
+    }
+
+    if ( data.type !== "Item" ) return false;
+    const item = await Item.implementation.fromDropData(data);
+
+    // Ensure the dropped item is a subclass
+    if ( item.type !== "subclass" ) {
```

Comment:
Should we also check `identifier` here? If you're dragging a subclass from outside the Compendium Browser (because you created it yourself maybe), then it will always be accepted as a valid subclass for the class because we override the existing identifier in `SubclassAdvancement#apply`.

Reply:
I didn't want to fully block support for "Universal Subclasses" (something WotC tried with the Strixhaven UA but didn't stick with).

Reply:
Would such subclasses not have a blank identifier?